From: Jan Grulich <jgrulich@redhat.com>
Date: Thu, 15 Jun 2023 12:41:00 +0000
Subject: Bug 1724900: WebRTC backport: PipeWire video capture - set device
 unique ID during initialization r=pehrsons,webrtc-reviewers

This is a simple backport of an WebRTC upstream change.

Upstream commit: 4beafa38d546ab6c0bb423c12762f0c4568aa5ce

Differential Revision: https://phabricator.services.mozilla.com/D176626
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/583ba187d9e5e6d5174a728936b7bf6afc5ba326
---
 modules/video_capture/linux/video_capture_pipewire.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules/video_capture/linux/video_capture_pipewire.cc b/modules/video_capture/linux/video_capture_pipewire.cc
index c1e6aae5ea..99109e00e6 100644
--- a/modules/video_capture/linux/video_capture_pipewire.cc
+++ b/modules/video_capture/linux/video_capture_pipewire.cc
@@ -62,6 +62,10 @@ int32_t VideoCaptureModulePipeWire::Init(const char* deviceUniqueId) {
 
   node_id_ = id.value();
 
+  const int len = strlen(deviceUniqueId);
+  _deviceUniqueId = new (std::nothrow) char[len + 1];
+  memcpy(_deviceUniqueId, deviceUniqueId, len + 1);
+
   return 0;
 }
 
-- 
2.34.1

